config/application.rb

```diff
@@ -100,6 +100,30 @@ class Foreman::Consoletie < Rails::Railtie
 
 module Foreman
   class Application < Rails::Application
+    config.load_defaults '6.1'
+
+    # Rails 5.0 changed this to true, but a lot of code depends on this
+    config.active_record.belongs_to_required_by_default = false
+
+    # Rails 5.1 changed this to false, re-enabling this due to https://github.com/theforeman/foreman/pull/9711/files#r1247901552
+    config.assets.unknown_asset_fallback = true
```

Comment:
I've also added this one, see https://github.com/theforeman/foreman/pull/9711/files#r1247901552